test/unittests/test_cpp.cpp

```diff
@@ -120,3 +124,36 @@ TEST(cpp, host_call)
 
     example_host_destroy_context(host_context);
 }
+
+TEST(cpp, result_raii)
+{
+    static auto release_called = false;
+    release_called = false;
+    auto release_fn = [](const evmc_result*) noexcept { release_called = true; };
+
+    {
+        auto raw_result = evmc_result{};
+        raw_result.status_code = EVMC_INTERNAL_ERROR;
+        raw_result.release = release_fn;
+
+        auto raii_result = evmc::result{raw_result};
+        raii_result.gas_left = -1;
+
+        auto raw_result2 = raii_result.raw();
+        EXPECT_EQ(raw_result2.status_code, EVMC_INTERNAL_ERROR);
+        EXPECT_EQ(raw_result2.gas_left, -1);
```

Comment:
For completeness can you also check for `gas_left` above for `raw_result` that it equals to `0`?

Reply:
According to my spec, it is undefined what the value of `raw_result.gas_left` would be. Here it will be still `-1` because we only zero the `.release`.

Reply:
I was considering zeroing it, but there are other pitfalls related to using "move-like" designs that we cannot easily eliminate.

Reply:
It is implicitly zero by memory initialization, isn't it? I'm talking about this test case here.